cpp/CMakeLists.txt

```diff
@@ -237,6 +237,7 @@ add_library(
   src/column/column_factories.cpp
   src/column/column_factories.cu
   src/column/column_view.cpp
+  src/column/dremel.cu
```

Comment:
I see that `dremel.hpp` is in `detail/utilities/` but its source file is in `column/`. This is inconsistent.

Reply:
Yes this bothered me before as well but I forgot to change it. I'm going to move the header to column since it's fundamentally a (list) column encoding API.

Reply:
Hmm I thought we had `include/column/detail/`, but apparently not. I think the best two options are to:
- Create `column/detail` and move this there.
- Move the header into `include/lists/detail`.
What do you think? Do you have a third option that you prefer?

Reply:
`dremel` is used only to compare lists, isn't it? I'm not sure.

If so, can we move it into the same place as `row_operators.cuh`?

Reply:
`dremel` is also used for Parquet encoding of lists, so it's not just for comparators. That's why I proposed `column/detail`, it's a generic column encoding for lists that we may even find other uses for in the future if someone is clever about new algorithms!

Reply:
Humnn, I'm reluctant to let it go into `column/*` because it's not a member of column class. So moving into `lists/detail/` seems more reasonable for me.

Reply:
Done.